app/assets/javascripts/omni_nav/google-custom-search.js

```diff
@@ -0,0 +1,314 @@
+var GoogleCustomSearch = (function() {
+  var GCS_ENGINE_ID = '015856566681218627934:2ndbiubovo4';
+  var GCS_SOURCE = location.protocol + '//www.google.com/cse/cse.js?cx=' + GCS_ENGINE_ID;
+
+  var $omniNav,
+      searchAPI;
+
+  var initialize = function(omniNavId, primarySearchId, utilitySearchId) {
+    $omniNav = $('#' + omniNavId);
+
+    // Google Search JavaScript API
+    // Source: https://developers.google.com/custom-search/docs/element#javascript
+    window.__gcse = {
+      parsetags: 'explicit',    // Components are rendered only with explicit calls
+      callback: onGoogleSearchInitialized
+    };
+
+    loadGoogleSearchIfNotLoaded();
+
+    searchAPI = {
+      primaryNav: new SearchComponent(primarySearchId),
+      utilityNav: new SearchComponent(utilitySearchId)
+    };
+    return searchAPI;
+  };
+
+  var loadGoogleSearchIfNotLoaded = function() {
+    // GCS is also loaded in omni-nav.js. Only load script if it doesnt exist already.
+    var alreadyLoaded = $('script[src="'+ GCS_SOURCE +'"]').length > 0;
+    if ( ! alreadyLoaded ) {
+      (function(){
+        var cx = GCS_ENGINE_ID;
+        var gcse = document.createElement('script');
+        gcse.type = 'text/javascript';
+        gcse.async = true;
+        gcse.src = GCS_SOURCE;
+        var s = document.getElementsByTagName('script')[0];
+        s.parentNode.insertBefore(gcse, s);
+      })();
+    }
+  }
+
+  /*
+   * Module initializes two forms, one for mobile and one for non-mobile. This works,
+   * but because one element in Google-generated part of the markup has a hardcoded
+   * ID, it's raising an alert in SiteImprove.
+   *
+   * TODO: create only one search component object and require OmniNav to toggle its
+   * position in the navbar based on device view.
+   */
+  var onGoogleSearchInitialized = function() {
+    searchAPI.primaryNav.init();
+    searchAPI.utilityNav.init();
+  };
+
+  /*
+   * This object represents the Search component (search form and results block). It wraps
+   * the Google element[0] and jQuery $elements representing the component in order to
+   * provide an API tailored to OmniNav.
+   *
+   * [0] https://developers.google.com/custom-search/docs/element#cse-element
+   */
+  var SearchComponent = function(containerId) {
```

Comment:
This replaces the `TwoColumnGCS` module:

https://github.com/chapmanu/cascade-assets/blob/2758afd76755884feb074b9abdafbe394ad05b93/app/assets/javascripts/omni-nav-v2.js#L384

I've renamed it to try to clarify its role and also tried to slim down the code a bit. For example, I eliminated as many module vars (under `// Internal Attrs`) as I could.